nannyml_cloud_sdk/model_evaluation/model.py

```diff
@@ -0,0 +1,372 @@
+import datetime
+import functools
+from typing import Optional, List, Dict
+
+import pandas as pd
+from frozendict import frozendict
+from gql import gql
+
+from nannyml_cloud_sdk._typing import TypedDict
+from nannyml_cloud_sdk.client import execute
+from nannyml_cloud_sdk.data import DATA_SOURCE_SUMMARY_FRAGMENT, DATA_SOURCE_EVENT_FRAGMENT, Data, DataSourceSummary, \
+    DataSourceFilter, DataSourceEvent
+from nannyml_cloud_sdk.enums import ProblemType, PerformanceMetric
+from nannyml_cloud_sdk.errors import InvalidOperationError
+from nannyml_cloud_sdk.model_evaluation.enums import HypothesisType
+from nannyml_cloud_sdk.model_evaluation.run import RunSummary, RUN_SUMMARY_FRAGMENT
+from nannyml_cloud_sdk.model_evaluation.schema import ModelSchema
+from nannyml_cloud_sdk.schema import normalize
+
+
+class ModelSummary(TypedDict):
+    """Summary of a model.
+
+    Attributes:
+        id: Unique identifier of the model (generated by NannyML Cloud when a model is created).
+        name: User-defined name of the model.
+        problemType: Type of problem the model is trying to solve.
+        createdAt: Timestamp when the model was created.
+    """
+    id: str
+    name: str
+    problemType: ProblemType
+    createdAt: datetime.datetime
+
+
+class ModelDetails(ModelSummary):
+    """Detailed information about a model.
+
+    Attributes:
+        latestRun: The currently active run or latest run performed for the model. This is ``None`` if no runs have been
+            performed yet.
+        nextRun: The next run scheduled for the model. This is ``None`` if there is a run currently active.
```

Comment:
The `nextRun` isn't available for evaluation models I believe, since we're currently not using scheduling for the evaluation module.